gwconfig/provider.go

```diff
@@ -0,0 +1,14 @@
+package gwconfig
+
+// gwconfig.Provider is an interface for configuring generic-worker.
+// Implementors include AWSProvisioner, AWSProvider, GCPConfigProvider, and
+// gwconfig.File for configuring generic-worker from a local file.
+type Provider interface {
```

Comment:
Maybe this should be ConfigProvider or ConfigSource to avoid confusion with worker-manager or worker-runner providers?

Reply:
Funnily enough in my first draft I called it ConfigProvider, but then I changed it based on [this advice](https://blog.golang.org/package-names):

> __Avoid stutter__. Since client code uses the package name as a prefix when referring to the package contents, the names for those contents need not repeat the package name. The HTTP server provided by the http package is called Server, not HTTPServer. Client code refers to this type as http.Server, so there is no ambiguity. 

Anything using Provider refers to it as `gwconfig.Provider` so hopefully it should be clear that it is a config provider. But I also wasn't totally happy with "Provider" either as it sounds like a Worker Manager term.